klayout/drc/rule_decks/mim_b.drc

```diff
@@ -0,0 +1,149 @@
+  mimtm_via = top_via.overlapping(mimtm_virtual)
+  mimtm2_l1 = mimtm_via.enclosed(topmin1_metal, 0.4.um, euclidian).polygons(0.001.um)
+  mimtm2_l2 = mimtm_via.not_outside(topmin1_metal).not(topmin1_metal)
+  mimtm2_l  = mimtm2_l1.join(mimtm2_l2)
+  mimtm2_l.output('MIMTM.2', "MIMTM.2 : Minimum MiM bottom plate overlap of Vian-1 layer.
+                   [This is applicable for Vian-1 within 1.06um oversize of FuseTop layer
+                   (referenced to virtual bottom plate)]: 0.4µm")
+  mimtm2_l1.forget
+  mimtm2_l2.forget
+  mimtm2_l.forget
+  mimtm_via.forget
+
+  # Rule MIMTM.3: Minimum MiM bottom plate overlap of Top plate is 0.6um.
+  logger.info('Executing rule MIMTM.3')
+  mimtm3_l1 = fusetop.enclosed(mimtm_virtual, 0.6.um).polygons(0.001.um)
+  mimtm3_l2 = fusetop.not_inside(mimtm_virtual)
+  mimtm3_l = mimtm3_l1.join(mimtm3_l2)
+  mimtm3_l.output('MIMTM.3', 'MIMTM.3 : Minimum MiM bottom plate overlap of Top plate: 0.6um')
+  mimtm3_l.forget
+  mimtm3_l1.forget
+  mimtm3_l2.forget
+  mimtm_virtual.forget
+
+  # Rule MIMTM.4: Minimum MiM top plate (FuseTop) overlap of Vian-1 is 0.4µm.
+  logger.info('Executing rule MIMTM.4')
+  mimtm4_l1 = fusetop.enclosing(top_via, 0.4.um, euclidian).polygons(0.001.um)
+  mimtm4_l2 = top_via.not_outside(fusetop).not(fusetop)
+  mimtm4_l  = mimtm4_l1.join(mimtm4_l2)
+  mimtm4_l.output('MIMTM.4', 'MIMTM.4 : Minimum MiM top plate (FuseTop) overlap of Vian-1: 0.4µm')
+  mimtm4_l1.forget
+  mimtm4_l2.forget
+  mimtm4_l.forget
+
+  # Rule MIMTM.5: Minimum spacing between top plate and the Vian-1 connecting to the bottom plate is 0.4µm.
+  logger.info('Executing rule MIMTM.5')
+  mimtm5_l1 = fusetop.separation(top_via.interacting(topmin1_metal), 0.4.um, euclidian)
+  mimtm5_l1.output('MIMTM.5',
+                   'MIMTM.5 : Minimum spacing between top plate and the Vian-1 connecting to the bottom plate: 0.4µm')
+  mimtm5_l1.forget
+
+  # Rule MIMTM.6: Minimum spacing between unrelated top plates is 0.6µm.
+  logger.info('Executing rule MIMTM.6')
+  mimtm6_l1 = fusetop.space(0.6.um, euclidian)
+  mimtm6_l1.output('MIMTM.6', 'MIMTM.6 : Minimum spacing between unrelated top plates: 0.6µm')
+  mimtm6_l1.forget
+
+  # Rule MIMTM.7: Min FuseTop enclosure by CAP_MK is 0um.
+  logger.info('Executing rule MIMTM.7')
+  mimtm7_l1 = fusetop.not(cap_mk)
+  mimtm7_l1.output('MIMTM.7', 'MIMTM.7 : Min FuseTop enclosure by CAP_MK: 0um')
+  mimtm7_l1.forget
+
+  # Rule MIMTM.8a: Minimum MIM cap area (defined by FuseTop area) is 25µm².
+  logger.info('Executing rule MIMTM.8a')
+  mimtm8a_l1 = fusetop.with_area(nil, 25.um)
+  mimtm8a_l1.output('MIMTM.8a', 'MIMTM.8a : Minimum MIM cap area (defined by FuseTop area): 25µm²')
+  mimtm8a_l1.forget
+
+  # Rule MIMTM.8b: Maximum single MIM Cap area (Use multiple MIM caps in
+  ## parallel connection if bigger capacitors are required) (um2) is 10000µm.
+  logger.info('Executing rule MIMTM.8b')
+  mimtm8b_l1 = fusetop.with_area(10_000.um, nil).not_in(fusetop.with_area(10_000.um))
```

Comment:
do we actually need the `not` there? as `with_area(10_000.um, nil)` would already return things that are over 10000um? or do you want to only print the edge that are outside of the area in the result?

Reply:
@proppy

`fusetop.with_area(10_000.um, nil)`, this will return polygons with area greater than or equal to 10000um.

So we have to exclude shapes with area equal to 10000um to get only ones greater than this value.